Lib/typing.py

```diff
@@ -1578,7 +1578,7 @@ def __init__(self, origin, metadata):
         if isinstance(origin, _AnnotatedAlias):
             metadata = origin.__metadata__ + metadata
             origin = origin.__origin__
-        super().__init__(origin, origin, name="Annotated")
+        super().__init__(origin, origin)
```

Comment:
It restores the past issue: `__name__` returning non-string.

I have a patch with other solution of this issue. After merging it your PR would not cause problems.

Reply:
See #27861.